### PR TITLE
Use RecyclableMemoryStream for MemoryStreams

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- RecyclableMemoryStream for MemoryStreams

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs [hunk 1]
```diff
@@ -168,7 +169,7 @@ void SerializeTimeoutException(IJsonRpcService service, IBufferWriter<byte> resu
                         long totalResponseSize = 0;
                         await foreach (JsonRpcResult result in jsonRpcProcessor.ProcessAsync(request, jsonRpcContext))
                         {
-                            Stream stream = jsonRpcConfig.BufferResponses ? new MemoryStream() : null;
+                            using Stream stream = jsonRpcConfig.BufferResponses ? RecyclableStream.GetStream() : null;
```
[the remark this answers is not in the document]
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs:172]: isn't it multiple dispose?

> Always dispose of each stream exactly once.

**Review by LukaszRozmej — COMMENTED** [review]
> Set the BlockSize, LargeBufferMultiple, MaximumBufferSize, MaximumLargePoolFreeBytes and MaximumSmallPoolFreeBytes properties to reasonable values for your application and resource requirements. Important!: If you do not set MaximumFreeLargePoolBytes and MaximumFreeSmallPoolBytes there is the possibility for unbounded memory growth!

File: src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (benaadams) [reply src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs:172]: Where is second dispose?

[the remark this answers is not in the document]
Comment (benaadams) [reply src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs:172]: > Does using work correctly here with null?

Yes `using` acts as `?.Dispose()`

**Comment by LukaszRozmej** [comment]
`System.ObjectDisposedException : Cannot access a disposed object.` some tests are failing

**Comment by benaadams** [comment]
> `System.ObjectDisposedException : Cannot access a disposed object.` some tests are failing

`leaveOpen: true` 🙄

**Comment by LukaszRozmej** [comment]
lets use it for recorder though!

**Comment by benaadams** [comment]
> lets use it for recorder though!

Added back to recorder, but lots of whitespace changes to add a new try/finally block (can't dispose the stream till much later)

File: src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs [hunk 2]
```diff
@@ -116,161 +117,169 @@ private JsonRpcRequest DeserializeObject(JsonElement element)
 
     public async IAsyncEnumerable<JsonRpcResult> ProcessAsync(PipeReader reader, JsonRpcContext context)
     {
-        reader = await RecordRequest(reader);
+        Stream? stream = null;
+        (reader, stream) = await RecordRequest(reader);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs:121]: Can you just dispose the reader and it would dispose the stream?

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**